GetNOTAM.py

```diff
@@ -13,15 +13,18 @@
 #getNotam: takes the lat, long, start and end time and the page number then runs an API call to the FAA for a json of the api
 #@returns parsed_req: the json of the request
 def getNotam(effectiveStartDate, effectiveEndDate, longitude, latitude, pageNum, radius):
-  # Convert datetime strings to the desired format
-  effectiveStartDate = datetime.strptime(effectiveStartDate, "%Y-%m-%d %H:%M:%S").strftime("%Y-%m-%dT%H:%M:%SZ")
-  effectiveEndDate = datetime.strptime(effectiveEndDate, "%Y-%m-%d %H:%M:%S").strftime("%Y-%m-%dT%H:%M:%SZ")  
+  # Convert start and end dates to the desired format if provided
+  if effectiveStartDate:
+    effectiveStartDate = datetime.strptime(effectiveStartDate, "%Y-%m-%d %H:%M:%S").strftime("%Y-%m-%dT%H:%M:%SZ")
+        
+  if effectiveEndDate:
+    effectiveEndDate = datetime.strptime(effectiveEndDate, "%Y-%m-%d %H:%M:%S").strftime("%Y-%m-%dT%H:%M:%SZ") 
   
   url = 'https://external-api.faa.gov/notamapi/v1/notams'
   url = (f"{url}?responseFormat=geoJson&effectiveStartDate={effectiveStartDate}"
        f"&effectiveEndDate={effectiveEndDate}&locationLongitude={longitude}"
        f"&locationLatitude={latitude}&locationRadius={radius}"
-       f"&pageNum={pageNum}&pageSize=50"
+       f"&pageNum={pageNum}&pageSize=1000"

```

Comment:
I think there is an assumption that requesting fewer but larger pages is faster, or is this verified?  is there an issue/ticket in the backlog to maybe test that assumption or figure out if there is an optimal page size?